operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -697,18 +747,115 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
+        if ((extraResources != null && extraResources.contains(subResourceType.value())) &&
+                (allowedIncludes == null || allowedIncludes.contains(parameterName)
+                || allowedIncludes.contains(simplifiedParameterName)) && 
+                searchParam != null) {
             searchParameters.add("_include", parameterName);
         } else {
             if (LOG.isLoggable(Level.FINE)) {
                 LOG.fine("Filtering out searchParameter because it is not supported by the server config: " + parameterName);
             }
         }
     }
+    
+    /**
+     * Read additional associated resources for the "extra" types (like location) 
+     * supported by this server configuration
+     *
+     * @param compartmentMemberType the type of resource currently drilling down on
+     * @param newEntries a list of entries of that compartment type that were found in a search
+     * @param allEntries a list of all entries that have been found so far (minus duplicates)
+     * @param resourceHelper an instance of FHIRResourceHelpers to use to make the calls
+     * @param extraResources a list of "extra" resource types configured for this server 
+     */
+    private void readsOfAdditionalAssociatedResources(String compartmentMemberType, List<Entry> newEntries,
+                List<Entry> allEntries, List<String> resourceIds, 
+                FHIRResourceHelpers resourceHelper, List<String> extraResources) throws Exception {
+        if (extraResources == null || extraResources.isEmpty()) {
+            return;
+        }
+        String url = FHIRRequestContext.get().getOriginalRequestUri().substring(0, FHIRRequestContext.get().getOriginalRequestUri().indexOf("/Patient/"));
+        List<Resource> resourceList = new ArrayList<Resource>();
+        for (Entry entry: newEntries) {
+            // Look up entries by reference
+            List<Reference> references = ReferenceFinder.getReferences(entry.getResource());
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, compartmentMemberType);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String externalId = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    String[] splitString = externalId.split("/");
+                    String type = splitString[0];
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, splitString[1], false, false, null, null).getResource());
```

Comment:
i wonder if it would speed things up to batch these reads up.  maybe add a comment as a spot for potential future enhancement?

Reply:
Done.